### ISSUE TITLE
Discrepency between your code implementation and paper

### ISSUE THREAD

**Issue by PosoSAgapo** [Neutralzz/syntax_aware_local_attention#1]
Hi, your work on incorporating syntax into Bert is very interesting and I am trying to develop your code. I noticed a discrepancy between your paper and your code:
In your paper, the Syntax Attention is` **softmax(QK^T/sqrt(d)+M)**`, which the _**M is the dependency mask matrix**_. However, your code seems does not work in this way. In your code, the dependency matrix is used to calculate a `**scaled_attention_mask** `that is used to _**multiply with the original attention matrix**_ in the BertSelfAttention layer, then the multiplied matrix is added with a normal attention mask that is not the dependency matrix mask. This makes it different from the method illustrated in the paper.
Did I understand your code in the wrong way? Or your code made changes after the paper was published?

**Comment by Neutralzz** [comment]
Please checkout the latest codes, which are consistent with our paper.

I'm sorry that previously released codes are another experiment, where the input matrix is the dependency distances, `DepAttentionScale` module predicts a distance threshold for each attention head and token, then computes the mask matrix `scaled_attention_mask` (which is similar to an 01 matrix). So the operation with attention matrix is multiplication. The following adding operation is to limit interaction with `[PAD]` tokens. These codes also perform well on several datasets (such as FCE), but can't obtain consistent gains on all datasets.
